### PR TITLE
WIP - DON'T MERGE - Declare Endpoints using ConductR ACLs

### PR DESCRIPTION
**Pull request by fsat**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -20,6 +20,19 @@ lazy val friendImpl = project("friend-impl")
       lagomJavadslTestKit
     )
   )
+  .settings(
```
Comment (markusjura) [inline build.sbt:23]: These settings should be automatically set by `sbt-lagom-bundle`. It shouldn't be necessary to set any kind of settings for a Lagom service.

Comment (huntc) [inline build.sbt:23]: I concur.

File: build.sbt [hunk 2]
```diff
@@ -76,11 +115,24 @@ lazy val frontEnd = project("front-end")
...
+    BundleKeys.endpoints := Map(
+      "chirperweb" -> Endpoint("http", 0, "chirperweb",
+        RequestAcl(
+          Http(
+            "^/chirper".r
+          ))
+      )
+    ),
+    javaOptions in Bundle ++= Seq(
+      "-Dapplication.context=/chirper/",
```
Comment (markusjura) [inline build.sbt:132]: Why do we need to override `application.context`?

Comment (huntc) [inline build.sbt:132]: You wouldn't have to given generated acls. Nice to see how it is done though. :-)

**Comment by markusjura** [comment]
General note: We should keep the activator template as simple as possible because this is the entrypoint for a lot of folks which are not used to Play or Lagom. In particular, if we can avoid overwriting `application.context` then we shouldn't do it. I consider this as an advanced Play feature.

**Comment by markusjura** [comment]
Ideally this activator template shouldn't be necessary to change due to the ACL support, i.e. We should try hard changing `sbt-lagom-bundle` / `sbt-bundle` such that no PR for a Lagom project is necessary.

**Comment by fsat** [comment]
@markusjura  and @huntc - thanks for the review. I should've mentioned earlier that I made this PR to made sure that Lagom does work with the ACL DSLs, and I'm not intending for this PR to be merged.

If I can make the all the changes in the `sbt-lagom-bundle`, I will close this PR (I'm still grateful for the review though!)

So far the services are pretty straightforward to make everything work.

The frontends are proving to be challenging with the URLs.

I've tried mapping path starting with `/chirper` rewritten to `/`, but this means the contents coming from the frontend requires some change so the reverse routing required by the links and the reactive.js routes would be correct.

The least intrusive option so far seems like to use Play's application context since this would mean I can move the context path of the application from `/` to `/chirper`. I agree, it's not great because the end user is affected by this. At this moment, I haven't got any idea on how to do this seamlessly for the framework's user.

I'm not sure if I should mount the chirper frontend with `/` since only 1 endpoint can be mapped with `/`. I'll also need to confirm in the ConductR code what would happen if an endpoint is declared with path beginning with `/` (i.e. would it ignore all other ACLs or not)

Hopefully this clarifies the reasoning behind this PR.

**Comment by huntc** [comment]
Thanks Felix. The frontend acl will require a declaration as it does now for the services uri. The acl should be for '/'. Note that the front is not a Lagom service, but a Play app. sbt-Lagom-bundle will not touch it. 

The other projects are Lagom services and should require no change. I recommend modifying sbt-lagom-bundle so that it presents acls. This will then provide an E2E test for you.

**Comment by markusjura** [comment]
Thanks @fsat to giving us some context. Agree with @huntc comment.

Another note in this respect: We will eventually create an additional `sbt-play-bundle` that can automate certain things for Play. We've already created the issue https://github.com/typesafehub/sbt-lagom-bundle/issues/14 for it. This bundle could then pick up the endpoints from the routes will and automatically set the ACLs for ConductR. I'll add that feature to the issue.

**Comment by TimMoore** [comment]
@fsat is there still a reason to keep this pull request around, or should we close it?

**Comment by fsat** [comment]
Closing this. ACLs is a feature of ConductR 2, and only when ConductR 1.1 is not supported we can work on this.

**Pull request closed without merging**